api/storage/version_endpoint_storage.go

```diff
@@ -50,9 +52,18 @@ func (v *versionEndpointStorage) Get(uuid uuid.UUID) (*models.VersionEndpoint, e
 }
 
 func (v *versionEndpointStorage) Save(endpoint *models.VersionEndpoint) error {
+	checkDBContraint(endpoint)
 	return v.db.Save(&endpoint).Error
 }
 
+func checkDBContraint(endpoint *models.VersionEndpoint) {
```

Comment:
nit: since this function not only check the endpoint but also alter the endpoint, I prefer to rename it to something like `sanitizeEndpoint`